src/codec.rs

```diff
@@ -131,6 +131,14 @@ pub trait Input {
 		self.read(&mut buf[..])?;
 		Ok(buf[0])
 	}
+
+	/// Descend into nested structure when decoding
+	fn descend(&mut self) -> Result<(), Error> {
```

Comment:
From the api docs and the usage I don't understand when we enter a nested structure.

If I have a struct as field of a struct, aren't we entering a new nested structure as well?

Reply:
It only makes sense to trace nested references as nested structures are not allowed in rust anyway. I'll update docs.